src/chain/chains.rs

```diff
@@ -0,0 +1,71 @@
+use std::ops::{Index, IndexMut};
+
+use super::chain::{Chain, ChainId};
+use super::pass::PassLinks;
+use resource::{Resource, Usage};
+
+/// Type alias for `Chain` bound to specific resource type.
+pub type ResourceChain<R: Resource, S, W> = Chain<R::Access, R::Layout, R::Usage, S, W>;
```

Comment:
I feel incoming Rust warnings about type generics not being enforced. That doesn't mean this code needs fixing, but rather - Rust needs to be.